src/vimageHosting.h

```diff
@@ -0,0 +1,123 @@
+#ifndef VGITHUBIMAGEHOSTING_H
+#define VGITHUBIMAGEHOSTING_H
+
+#include <QObject>
+#include <QtNetwork>
+#include <QProgressDialog>
+#include <QDesktopServices>
+#include <QUrl>
+#include <QByteArray>
+#include <QNetworkAccessManager>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+#include <QApplication>
+#include <vfile.h>
+#include <QClipboard>
+
+class VGithubImageHosting : public QObject
+{
+    Q_OBJECT
+public:
+    explicit VGithubImageHosting(VFile *p_file, QWidget *parent = 0);
+    // github image hosting
+    // GitHub identity authentication
+    void authenticateGithubImageHosting(QString p_token);
+    // Upload a single image
+    void githubImageBedUploadImage(QString username,QString repository,QString image_path,QString token);
```

Comment:
- One space to separate parameters.
- Const reference.

```suggestion
    void githubImageBedUploadImage(const QString &p_username,
                                   const QString &p_repository,
                                   const QString &p_image_path,
                                   const QString &p_token);
```